recipes/stlab/all/conanfile.py

```diff
@@ -1,180 +1,218 @@
-from conans import ConanFile, tools
-from conans.tools import Version
-from conans.errors import ConanInvalidConfiguration
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import copy, get, rmdir
+from conan.tools.scm import Version
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.microsoft import check_min_vs, is_msvc
+
 import os
 
+required_conan_version = ">=1.52.0"
+
 class Stlab(ConanFile):
     name = 'stlab'
     description = 'The Software Technology Lab libraries.'
     url = 'https://github.com/conan-io/conan-center-index'
     homepage = 'https://github.com/stlab/libraries'
     license = 'BSL-1.0'
-    topics = 'conan', 'c++', 'concurrency', 'futures', 'channels'
+    topics = 'concurrency', 'futures', 'channels'
 
-    settings = "arch", "os", "compiler", "build_type", 
+    settings = "arch", "os", "compiler", "build_type",
 
     options = {
-        "boost_optional": [True, False],
-        "boost_variant": [True, False],
-        "coroutines": [True, False],
+        "with_boost": [True, False],
+        "no_std_coroutines": [True, False],
+        "future_coroutines": [True, False],
         "task_system": ["portable", "libdispatch", "emscripten", "pnacl", "windows", "auto"],
+        "thread_system": ["win32", "pthread", "pthread-emscripten", "pthread-apple", "none", "auto"],
     }
 
     default_options = {
-        "boost_optional": False,
-        "boost_variant": False,
-        "coroutines": False,
+        "with_boost": False,
+        "no_std_coroutines": True,          #TODO: how to make checks similar to what are made in Cmake https://github.com/stlab/libraries/blob/main/cmake/StlabUtil.cmake#L35
+        "future_coroutines": False,
         "task_system": "auto",
+        "thread_system": "auto",
     }
 
-    no_copy_source = True
-    _source_subfolder = 'source_subfolder'
+    short_paths = True
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 17
 
-    def _use_boost(self):
-        return self.options.boost_optional or self.options.boost_variant
+    @property
+    def _compilers_minimum_version(self):
+        return {"gcc": "9",
+                "Visual Studio": "15.8",
```

Comment:
https://github.com/conan-io/conan-center-index/pull/13048/files#diff-c4747a2c3a2f1b6ca9f4b271de4f2d8958e72c24e71f92a4c3a8bc96f460c385R147

This is not needed

```suggestion
```